tedana/tests/test_integrity _robustica.py

```diff
@@ -0,0 +1,667 @@
+from pkg_resources import resource_filename
+
+from tedana.io import InputHarvester
+from tedana.workflows import t2smap as t2smap_cli
+from tedana.workflows import tedana as tedana_cli
+from tedana.workflows.ica_reclassify import ica_reclassify_workflow
+
+# Need to see if a no BOLD warning occurred
+LOGGER = logging.getLogger(__name__)
+# Added a testing logger to output whether or not testing data were downlaoded
+TestLGR = logging.getLogger("TESTING")
+
+
+def check_integration_outputs(fname, outpath, n_logs=1):
+    """
+    Checks outputs of integration tests.
+
+    Parameters
+    ----------
+    fname : str
+        Path to file with expected outputs
+    outpath : str
+        Path to output directory generated from integration tests
+    """
+
+    # Gets filepaths generated by integration test
+    found_files = [
+        os.path.relpath(f, outpath)
+        for f in glob.glob(os.path.join(outpath, "**"), recursive=True)[1:]
+    ]
+
+    # Checks for log file
+    log_regex = "^tedana_[12][0-9]{3}-[0-9]{2}-[0-9]{2}T[0-9]{2}[0-9]{2}[0-9]{2}.tsv$"
+    logfiles = [out for out in found_files if re.match(log_regex, out)]
+    assert len(logfiles) == n_logs
+
+    # Removes logfiles from list of existing files
+    for log in logfiles:
+        found_files.remove(log)
+
+    # Compares remaining files with those expected
+    with open(fname, "r") as f:
+        expected_files = f.read().splitlines()
+    expected_files = [os.path.normpath(path) for path in expected_files]
+
+    if sorted(found_files) != sorted(expected_files):
+        expected_not_found = sorted(list(set(expected_files) - set(found_files)))
+        found_not_expected = sorted(list(set(found_files) - set(expected_files)))
+
+        msg = ""
+        if expected_not_found:
+            msg += "\nExpected but not found:\n\t"
+            msg += "\n\t".join(expected_not_found)
+
+        if found_not_expected:
+            msg += "\nFound but not expected:\n\t"
+            msg += "\n\t".join(found_not_expected)
+        raise ValueError(msg)
+
+
+def data_for_testing_info(test_dataset=str):
+    """
+    Get the path and download link for each dataset used for testing.
+
+    Also creates the base directories into which the data and output
+    directories are written
+
+    Parameters
+    ----------
+    test_dataset : str
+       References one of the datasets to download. It can be:
+        three-echo
+        three-echo-reclassify
+        four-echo
+        five-echo
+
+    Returns
+    -------
+    test_data_path : str
+       The path to the local directory where the data will be downloaded
+    osf_id : str
+       The ID for the OSF file.
+       Data download link would be https://osf.io/osf_id/download
+       Metadata download link would be https://osf.io/osf_id/metadata/?format=datacite-json
+    """
+
+    tedana_path = os.path.dirname(tedana_cli.__file__)
+    base_data_path = os.path.abspath(os.path.join(tedana_path, "../../.testing_data_cache"))
```

Comment:
Throughout much of the subsequent code, do not use "`/`" / "`../`" for generating filesystem paths. Use `os.path.join()`, `os.pardir` and `os.path.normpath()`.